src/models/SubProgram.js

```diff
@@ -22,5 +23,16 @@ SubProgramSchema.virtual('courses', {
   foreignField: 'subProgram',
 });
 
+function setIsStrictlyELearning() {
+  if (this.steps.length && this.steps[0].type) {
```

Comment:
this.steps && ...

Reply:
mon retour n'etait pas clair : `this.steps && this.steps.length && this.steps[0].type`

Reply:
done